app/controllers/status_controller.rb

```diff
@@ -1,8 +1,40 @@
 # frozen_string_literal: true
 
+require 'date'
+
 class StatusController < ApplicationController
+  include TokenUser
+
   def index
     render 'index', layout: false
   end
 
+  # /whoami returns basic information about the request client from Conjur's
+  # perspective, including a role_id from a provided access token (if provided)
+  # and the origin IP address of the request.
+  #
+  # This is useful for debugging authentication with access tokens and configuring
+  # proxies or load balancers.
+  def whoami
+    render json: { 
+      client_ip: request.ip,
+      user_agent: request.user_agent,
+    }.merge(token_attributes)
+  end
+
+  def token_attributes
+    if token_user?
+      {
+        account: token_user.account,
+        login: token_user.login,
+        token_issued_at: Time.at(token_user.token.claims["iat"])
```

Comment:
```suggestion
        token_issued_at: Time.at token_user.token.claims["iat"] || 'unknown issuing time'
```
https://github.com/cyberark/slosilo/blob/4121d4967fc0bc329d73d31e15ee8e9c31962832/lib/slosilo/key.rb#L148

Can this use-case be valid? failing to parse `iat` leasing us to publising stacktrace?

Reply:
No, the token is validated before the endpoint is reached. If the `iat` is missing, the token is invalid and the Rack middleware will return `unauthorized`. So we can be sure `iat` will be present when returning this response.